.github/release-checklist.md

```diff
@@ -0,0 +1,63 @@
+# Template to use for release PRs from `develop` to `main`
+
+:warning: **DO NOT MERGE (YET)** :warning:
+
+**Please **do** add approvals if you agree as otherwise we won't be able to release.**
+
+PR for tracking changes for the x.x.x release. Target release date: **DOW MONTH DAY YEAR**.
+
+## Release checklist
+
+### General
+
+- [ ] Verify, and if necessary, update the allowed version ranges for various dependencies in the `composer.json` - PR #xxx
+- [ ] PHPCS: check if there have been releases since the last WordPressCS release and check through the changelog to see if there is anything WordPressCS could take advantage of - PR #xxx
+- [ ] PHPCSUtils: check if there have been releases since the last WordPressCS release and update WordPressCS code to take advantage of any new utilities - PR #xxx
+- [ ] PHPCSExtra: check if there have been releases since the last WordPressCS release and check through the changelog to see if there is anything WordPressCS could take advantage of - PR #xxx
+- [ ] Check if the minimum WP version property needs updating in `MinimumWPVersionTrait::$default_minimum_wp_version` and if so, action it - PR #xxx
```

Comment:
We should note that we default to three versions behind current.

Reply:
That is documented in the property docblock in the trait. Do we really need to duplicate that information here ?